tests/Console/E2ETest.php

```diff
@@ -0,0 +1,243 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Console;
+
+use Composer\Autoload\ClassLoader;
+use Infection\Command\ConfigureCommand;
+use Infection\Console\Application;
+use Infection\Console\InfectionContainer;
+use Infection\Finder\ComposerExecutableFinder;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Process;
+
+class E2ETest extends TestCase
+{
+    const EXPECT_ERROR = 1;
+    const EXPECT_SUCCESS = 0;
+
+    private $cwd;
+
+    /**
+     * @var \Composer\Autoload\ClassLoader|null
+     */
+    private $previousLoader;
+
+    protected function setUp()
+    {
+        // Without overcommit this test fails with `proc_open(): fork failed - Cannot allocate memory`
+        if (strpos(PHP_OS, 'Linux') === 0 &&
+            is_readable('/proc/sys/vm/overcommit_memory') &&
+            file_get_contents('/proc/sys/vm/overcommit_memory') == 2) {
+            $this->markTestSkipped('This test needs copious amounts of virtual memory. It will fail unless it is allowed to overcommit memory.');
+        }
```

Comment:
I've spend several hours trying to find the culprit only to find that I changed this setting while debugging a Redis install long time ago.